lib/gyp/generator/ninja/index.js

```diff
@@ -140,6 +140,7 @@ Ninja.prototype.expand = function expand(p, productDir) {
     p = p.replace(/\$!INTERMEDIATE_DIR/g, intDir);
   }
 
+  p = p.replace(/(\.\.[\\\/])\$\|CONFIGURATION_NAME/g, `$1${this.configDir}`);
```

Comment:
The (faulty) use case in [icu-generic.gyp](https://github.com/nodejs/node/blob/master/tools/icu/icu-generic.gyp#L254) assumes it knows where the output is base on configuration and calles for
```
'-P', '../../<(CONFIGURATION_NAME)'
```
My line assumes that path manipulation (`../`) with `CONFIGURATION_NAME` actually want the configuration dir, not just name.
So we get `../../out/Release` and not just `../../Release`.